src/ompl/base/objectives/src/PathLengthOptimizationObjective.cpp

```diff
@@ -62,10 +64,13 @@ ompl::base::Cost ompl::base::PathLengthOptimizationObjective::motionCostHeuristi
     return motionCost(s1, s2);
 }
 
-#if OMPL_HAVE_EIGEN3
 ompl::base::InformedSamplerPtr ompl::base::PathLengthOptimizationObjective::allocInformedStateSampler(const ProblemDefinitionPtr probDefn, unsigned int maxNumberCalls) const
 {
-    // Make the direct path-length informed sampler and return
+    // Make the direct path-length informed sampler and return. If OMPL was compiled with Eigen, a direct version is available, if not a rejection-based technique can be used
+#if OMPL_HAVE_EIGEN3
     return boost::make_shared<PathLengthDirectInfSampler>(probDefn, maxNumberCalls);
-}
+#else
+    OMPL_INFORM("%s: Direct sampling of the path-length objective requires Eigen. As this version of OMPL was compiled without Eigen support, the informed sampling is defaulting to rejection sampling.", description_.c_str());
+    return boost::make_shared<RejectionInfSampler>(probDefn, maxNumberCalls);
```

Comment:
I understand that is valid, but even that preprocessor directive, that will be set to some default value and the user does not know of it. This is a one time notification that forces the user to allocate a different sampler, or to install Eigen. This I think is much better than silent  approximate fixes which may not be what the user wants.